src/parser/serializer.ts

```diff
@@ -806,6 +806,8 @@ export function serializeInto(
         index = serializeString(buffer, key, value, index, true);
       } else if (typeof value === 'number') {
         index = serializeNumber(buffer, key, value, index, true);
+      } else if (typeof value === 'bigint') {
+        throw new TypeError('Do not know how to serialize a BigInt, please use Decimal128');
```

Comment:
```suggestion
        throw new TypeError('Unsupported type BigInt, please use Decimal128');
```

this is just a deep nitpick, I think we do know how to serialize them, we just don't support them being passed in raw for now

Reply:
It just matches JSON's error precisely
But I think I agree and I'll change this